TF-5568 add support for project custom permissions (#745)
hashicorp/go-tfe: CHANGELOG.md

## Enhancements
* Added BETA support for including `projects` relationship and `projects-count` attribute to policy_set on create by @hs26gill [#737](https://github.com/hashicorp/go-tfe/pull/737)
* Added BETA method `AddProjects` and `RemoveProjects` for attaching/detaching policy set to projects by @Netra2104 [#735](https://github.com/hashicorp/go-tfe/pull/735)

> * Added BETA support for adding and updating custom permissions to `TeamProjectAccesses`. A `TeamProjectAccessType` of `"custom"` can set various permissions applied at

Reviewer: ```suggestion
* Added BETA support for adding and updating custom permissions to `TeamProjectAccess`. A `TeamProjectAccessType` of `"custom"` can set various permissions applied at
```

Reviewer: also yeah don't forget your credit! :)

> the project level to the project itself (`ProjectAccessOptions`) and all of the workspaces in a project (`WorkspaceAccessOptions`).[#745](https://github.com/hashicorp/go-tfe/pull/745)

Reviewer 2: ```suggestion
the project level to the project itself (`ProjectAccessPermissionsOptions`) and all of the workspaces in a project (`WorkspaceAccessPermissionsOptions`).[#745](https://github.com/hashicorp/go-tfe/pull/745)
```

> the project level to the project itself (`ProjectAccessOptions`) and all of the workspaces in a project (`WorkspaceAccessOptions`).[#745](https://github.com/hashicorp/go-tfe/pull/745)

Reviewer 3: I think when you rebase for some reason this changelog removes your github handle @rberecka

> the project level to the project itself (`ProjectAccessPermissionsOptions`) and all of the workspaces in a project (`WorkspaceAccessPermissionsOptions`).[#745](https://github.com/hashicorp/go-tfe/pull/745)

Reviewer 3: 💅 can you add your github handle before the PR link? @rberecka

> the project level to the project itself (`ProjectAccessPermissionsOptions`) and all of the workspaces in a project (`WorkspaceAccessPermissionsOptions`).[#745](https://github.com/hashicorp/go-tfe/pull/745)

Author: ```suggestion
the project level to the project itself (`ProjectAccessPermissionsOptions`) and all of the workspaces in a project (`WorkspaceAccessPermissionsOptions`) by @rberecka [#745](https://github.com/hashicorp/go-tfe/pull/745)
```
